cockatrice/src/gamesmodel.cpp

```diff
@@ -91,32 +96,98 @@ QVariant GamesModel::data(const QModelIndex &index, int role) const
-            return result.join(", ");
+        case DESCRIPTION: 
+            switch(role) {
+            case SORT_ROLE:
+            case Qt::DisplayRole:
+                return QString::fromStdString(g.description());
+            case Qt::TextAlignmentRole:
+                Qt::AlignLeft;
+            default:
+                return QVariant();
+        }
+        case CREATOR: {
+            switch(role) {
+            case SORT_ROLE:
+            case Qt::DisplayRole:
+                return QString::fromStdString(g.creator_info().name());
+            case Qt::DecorationRole:
+                switch(g.creator_info().user_level()) {
+                case 1: {
+                    QPixmap avatarPixmap = UserLevelPixmapGenerator::generatePixmap(13, (UserLevelFlags)g.creator_info().user_level());
+                    return QIcon(avatarPixmap);
+                        }
+                case 3:{
+                    QPixmap avatarPixmap = UserLevelPixmapGenerator::generatePixmap(13, (UserLevelFlags)g.creator_info().user_level());
+                    return QIcon(avatarPixmap);
+                       }
+                default:
+                    break;
+                }
+            default:
+                return QVariant();
+            }
         }
-        case 5: return g.with_password() ? ((g.spectators_need_password() || !g.spectators_allowed()) ? tr("yes") : tr("yes, free for spectators")) : tr("no");
-        case 6: {
-            QStringList result;
-            if (g.only_buddies())
-                result.append(tr("buddies only"));
-            if (g.only_registered())
-                result.append(tr("reg. users only"));
-            return result.join(", ");
+        case GAME_TYPE:
+            switch (role) {
+            case SORT_ROLE:
+            case Qt::DisplayRole: {
+                QStringList result;
+                GameTypeMap gameTypeMap = gameTypes.value(g.room_id());
+                for (int i = g.game_types_size() - 1; i >= 0; --i)
+                    result.append(gameTypeMap.value(g.game_types(i)));
+                return result.join(", ");
+                }
+            case Qt::TextAlignmentRole:
+                return Qt::AlignLeft;
+            default:
+                return QVariant();
         }
-        case 7: return QString("%1/%2").arg(g.player_count()).arg(g.max_players());
-        case 8: return g.spectators_allowed() ? QVariant(g.spectators_count()) : QVariant(tr("not allowed"));
+        case RESTRICTIONS: 
+            switch(role) {
+            case SORT_ROLE:
+            case Qt::DisplayRole: {
+                QStringList result;
+                if (g.only_buddies())
+                    result.append(tr("buddies only"));
+                if (g.only_registered())
+                    result.append(tr("reg. users only"));
+                if (g.with_password())
+                    result.append(tr("password"));
```

Comment:
Can you move the password test to be first so that the lock icon shows up next to it? Right now with multiple restrictions the icon is left and password text is right.